doc/developer/design/20231211_datadog_sink.md

```diff
@@ -0,0 +1,162 @@
+
+I discarded sending updates when a metric's value changes (streaming without state.) Collections like storage usage are updated once an hour, and this would make charts empty in Datadog for an hour or any smaller time range. This issue makes the need to have a state to keep submitting the same metric values even if they haven't changed. 
+
+### Datadog Metrics
+
+Submitting metrics to Datadog API requires the following fields:
+<details>
+<summary>Fields</summary>
+    
+| Field             | Type      | Definition|
+|-------------------|-----------|-----------|
+| metric [required] | String    | The name of the timeseries.                                                                                              |
+| points [required] | [Object]  | Points relating to a metric. All points must be objects with timestamp and a scalar value (cannot be a string). Timestamps should be in POSIX time in seconds, and cannot be more than ten minutes in the future or more than one hour in the past. |
+| resources         | [Object]  | A list of resources to associate with this metric.                                                                       |
+| source_type_name  | String    | The source type name.                                                                                                    |
+| tags              | [String]  | A list of tags associated with the metric.                                                                              |
+| type              | enum      | The type of metric. The available types are 0 (unspecified), 1 (count), 2 (rate), and 3 (gauge). Allowed enum values: 0, 1, 2, 3                                                        |
+| unit              | String    | The unit of point value.                                                                                                 |
+| interval          | Int64     | If the type of the metric is rate or count, define the corresponding interval.                                          |
+| metadata          | [Object]  | Metadata for the metric.                                                                                                 |
+    
+</details>
+
+All of the fields except the timestamp are set by the user. Materialize will set the timestamp before retrieving the metrics from its source.
+
+#### Timestamp
+
+In Materialize, we have two values to set the timestamp field: `mz_now()` and `now()`. I believe `now()` is enough. The integrations we recommend to our users today use the wall-clock time as a timestamp, so I don’t see any strong reason to switch to `mz_now()` unless we want correctness.
```

Comment:
In sink land I think it will actually be easier to use the logical time, and that's what I'd recommend doing here!